src/Sarif/CodeGenHints.json

```diff
@@ -1,12 +1,64 @@
 {
+  "annotatedCodeLocation": [
+    {
+      "kind": "BaseTypeHint",
+      "arguments": {
+        "baseTypeNames": [
+          "PropertyBagHolder"
+        ]
+      }
+    }
+  ],
   "AnnotatedCodeLocation.Properties": [
     {
-      "kind": "DictionaryHint"
+      "kind": "DictionaryHint",
+      "arguments": {
+        "valueTypeName": "SerializedPropertyInfo",
+        "namespaceName": "Microsoft.CodeAnalysis.Sarif.Readers",
```

Comment:
That is the namespace in which SerializedPropertyInfo resides. It causes the directive "using Microsoft.CodeAnalysis.Sarif.Readers" to appear at the top of any file that declares an IDictionary<string, SerializedPropertyInfo>. Otherwise you get a compiler error because it can't find SerializedPropertyInfo.

It would also work to specify "Microsoft.CodeAnalysis.Sarif.Readers.SerializedPropertyInfo" as the value of the "valueTypeName" argument, and to omit the "namespaceName" argument, but then the generated property declaration would have the type IDictionary<string,Microsoft.CodeAnalysis.Sarif.Readers.SerializedPropertyInfo>. I liked this better.

---
In reply to: [62769424](https://github.com/Microsoft/sarif-sdk/pull/257#discussion_r62769424) [](ancestors = 62769424,62749197)